ElementInternalsType/explainer.md

```diff
@@ -11,23 +11,257 @@
+- **Click event activation**: Fire click events when activated via mouse click, Enter key, Space key, or other activation methods
+- **Focusable by default**: The element becomes focusable and participates in tab navigation without requiring `tabindex`
+- **Default ARIA semantics**: Have an [button](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Reference/Roles/button_role) default ARIA role. Note ElementInternals.role is not changed. This is following the pattern where Element.role does not reflect native `<button>`'s `button` ARIA role.
+
+### Order of precedence regarding ARIA role
+The order is `<custom-button role=foo>` > `ElementInternals.role` > default `button` role  via `buttonActivationBehaviors`
+
+When `buttonActivationBehaviors = true`, the custom element's default ARIA role will become `button` and this will be the used role if no explicit role is specified by the author. If the author sets `elementInternals.role`, the value of `elementInternals.role` will be the used role, taking precedence over the default role. If the author sets the `role` attribute on the custom element, the value of the `role` attribute will be the used role, taking precedence over both `elementInternals.role` and the default role.
+
+### `buttonActivationBehaviors` does not change element appearance
+Setting `buttonActivationBehaviors` gives a custom element button activation behaviors, but the custom element's appearance does not change. In other words, the custom element does not take on default, author-specified or user-specified styles that target the native button element, since the custom element has a different tag name (e.g., `<fancy-button>` instead of `<button>`).
 
-## Proposal: add static `behavesLike` property with behavior-specific interface mixins
+### `buttonActivationBehaviors` and `formAssociated`
+If `buttonActivationBehaviors` is set to true, `static formAssociated = true/false;` becomes a no-op since the element will automatically gain form association capabilities from `buttonActivationBehaviors`.
 
-We propose enabling web component authors to create custom elements with native behaviors by adding a static `behavesLike` property to their custom element class definition. This property can be set to string values that represent native element types:
+## Examples
+
+### Custom button with popover invocation
+
+This example shows how to create a custom button that can invoke a popover element using the `commandfor` and `command` attributes:
+
+```js
+class CustomButton extends HTMLElement {
+    static buttonActivationBehaviors = true;
+}
+
+customElements.define('custom-button', CustomButton);
+```
+
+```html
+<custom-button commandfor="my-popover" command="toggle-popover">
+    Toggle the popover
+</custom-button>
+
+<div id="my-popover" popover>
+    <p>This popover is controlled by the custom button!</p>
+</div>
+```
+
+### Custom button with dialog invocation

```

Comment:
This example seems redundant with the previous one.
A way to potentially shorten the doc while still having a `<dialog>` example might be to eliminate this and change the "Custom button with imperative property configuration" example to use a `<dialog>` instead.